lib/que/job.rb

```diff
@@ -80,6 +84,14 @@ def self.run(*args)
     # have been passed in.
     def initialize(attrs)
       @attrs = attrs
+      @args = Array(attrs[:args])
+
+      @kwargs = if @args.last.is_a?(Hash) && @args.last.keys == [KWARGS_KEY]
```

Comment:
Unlikely to happen, but just one concern here is that when the args are stored in the DB we cannot differentiate between a genuine `Hash` argument `{ "__kwargs" => { foo: ... } }` and a keyword argument `foo:`; if the user adds a hash  `{ "__kwargs" => ... }` as the last argument to a job that does not expect keyword arguments, will that break things by trying to interpret it as kwargs?

Reply:
It would - I toyed with the idea of having a separate `kwargs` column in the DB but honestly they are just another kind of `arg` really so I'm trying to use the same array for all args. I picked the key `__kwargs` to make it _intensely_ unlikely that you would accidentally do this yourself, but there's nothing stopping you doing that! If you can think of an alternate way to represent this in the same array (e.g. using a sentinel like `[arg, arg, :KWARGS, {...}]` I'm open to suggestions since it's easy to change!

Reply:
Yeah that makes sense, ok with the current approach then. Would be nice if we could catch this as a rubocop cop, but I don't think it will be possible...